Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpAppxBuildTools.cs

```diff
@@ -408,6 +438,125 @@ public static void AddGazeInputCapability(IBuildInfo buildInfo)
             rootElement.Save(manifestFilePath);
         }
 
+        /// <summary>
+        /// Adds the 'Research Mode' capability to the manifest.
+        /// </summary>
+        /// <remarks>
+        /// This is only for research projects and should not be used in production.
+        /// For further information take a look at https://docs.microsoft.com/en-us/windows/mixed-reality/research-mode.
+        /// Note that this function is only public to poke a hole for testing - do not
+        /// take a dependency on this function.
+        /// </remarks>
+        public static void AddResearchModeCapability(XElement rootNode)
+        {
+            // Add rescap Namespace to package tag
+            XNamespace rescapNs = "http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities";
+            var rescapAttribute = rootNode.Attribute(XNamespace.Xmlns + "rescap");
+            if (rescapAttribute == null)
+            {
+                rescapAttribute = new XAttribute(XNamespace.Xmlns + "rescap", rescapNs);
+                rootNode.Add(rescapAttribute);
+            }
+
+            // Add rescap to IgnorableNamespaces
+            var ignNsAttribute = rootNode.Attribute("IgnorableNamespaces");
+            if (ignNsAttribute == null)
+            {
+                ignNsAttribute = new XAttribute("IgnorableNamespaces", "rescap");
+                rootNode.Add(ignNsAttribute);
+            }
+
+            if (!ignNsAttribute.Value.Contains("rescap"))
+            {
+                ignNsAttribute.Value += " rescap";
+            }
+
+            // If the capabilities container tag is missing, make sure it gets added.
```

Comment:
Nit: Consider moving this (and the lines in AddGazeInputCapability) to a help function, since they do the same thing (i.e. ensuring and then returning the capabilities element).

Alternatively, it could also be good to have a function that's like AddCapability(string capabilityType, string capabilityName), where you could call it like:

```csharp
AddCapability(rootNode.GetDefaultNamespace()+"DeviceCapability", "gazeInput");
AddCapability(rescapNs + "Capability", "perceptionSensorsExperimental");
```

This should reduce some of the copyiage that occurs here and would simplify stuff a bit

Reply:
Good idea. I will move this to a helper function.